clang/lib/Driver/Driver.cpp

```diff
@@ -4686,13 +4686,28 @@ class OffloadingActionBuilder final {
           else
             continue;
         } else if (A->getOption().matches(options::OPT_Xsycl_backend)) {
-          if (SYCLTripleList.size() > 1) {
+          // While the user may have provided a single AOT triple, the generic
+          // spir64 triple could have been added automatically by us upon
+          // discovering a spir64 section in one of the object/library inputs.
+          // To allow for automatic detection of the toolchain that needs to
+          // receive -Xsycl-target-* arguments, drop the autodetected spir64
+          // triple from the local copy of the targets list.
+          // FIXME: Is there a way to add the autodetected generic triple at
+          // a later stage instead?
```

Comment:
As this is a check of the offload specific list of triples, maybe we can add the auto-detected triple actions after we process the user defined targets in `SYCLTripleList`.  This would be manipulated in `appendLinkDependences`.  I'm OK with the exclusion method at this point though.

Reply:
Thanks for pointing me to `appendLinkDependences()`; since I'm still going to have to figure out the Linux crashes that are likely related to the exclusion, I'll attempt this method as well.

Reply:
Done, also updated the description. Turned out to be much easier than expected!